File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    answer = []
    
    left, right = 0, 0
    s = sequence[0]
    possible = []
    
    while True:
        if s < k:
            if (right + 1) < len(sequence):
                right += 1
                s += sequence[right]
            else:
                break
            
        elif s > k:
            s -= sequence[left]
            left += 1
            
        else:
            possible.append([left, right])
            s -= sequence[left]
            left += 1
            
    min_length = len(sequence) + 1
    
    for start, end in possible:
        if end - start + 1 < min_length:
            min_length = end - start + 1
            answer = [start, end]
        
    return answer

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int[] answer = new int[2];
        int left = 0;
        int right = 0;
        int current = sequence[0];
        int min_len = 10000000;
        
        while (left <= right) {
            if (current == k) {
                int len = right - left + 1;
                
                if (len < min_len) {
                    min_len = len;
                    answer[0] = left;
                    answer[1] = right;
                }
                
                current -= sequence[left];
                left++;
            } 
            
            else if (current < k) {
                if (right + 1 >= sequence.length) {
                    break;
                }
                
                right++;
                current += sequence[right];
            }
            
            else {
                current -= sequence[left];
                left++;
            }
        }
        
        return answer;
    }
}